### ISSUE TITLE
Meaningful Text Extraction

### ISSUE THREAD

**Issue by mbrown1413** [ITNG/yurika#68]
BeautifulSoup's `get_text()` doesn't understand the meaning of tags when extracting text.

```python
BeautifulSoup(
    '<div>he<b>ll</b>o<br>world</div><div>This is a block element</div>'
).get_text()
```
Returns: `helloworldThis is a block element`

Should be: `hello\nworld\nThis is a block element`

The problem is that:
  * `<div>` should be treated as a block element, and (for our purposes) separated by a newline.
  * `<br>` should be replaced with `\n`
  * `<b>` should be ignored (not replaced with a newline like `<div>` (this is already handled correctly)

Basically, we need to have 3 cases for tags: inline, block, and break. The HTML spec should give a list of common tags. I fixed the `<br>` issue in the "html-breaks" branch, but the others are still a problem. The code just needs cases added for inline and block, along with a list of elements that belong to each case (or just a list of inline, and treat all others as block, or vice versa).

For a previous discussion, see: https://bugs.launchpad.net/beautifulsoup/+bug/1768330
